homework/index.js

```diff
@@ -29,14 +29,116 @@
+        let selectedOption = document.getElementById('selectId').options[
+          document.getElementById('selectId').selectedIndex
+        ].text;
+        const repoDetailsDiv = createAndAppend('div', root, { id: 'repoDetailsDiv' });
+        const repoDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const descriptionDiv = createAndAppend('div', repoDetailsDiv, {
+          class: 'divClass',
+          id: 'descDiv',
+        });
+        const forkDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const updatedDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+
+        createAndAppend('span', descriptionDiv, {
+          id: 'descSpan',
+          text: 'Description:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', forkDiv, {
+          id: 'forkSpan',
+          text: 'Forks:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', updatedDiv, {
+          id: 'updateSpan',
+          text: 'Updated:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', repoDiv, { id: 'repoSpan', text: 'Repository:' });
+        createAndAppend('p', descriptionDiv, {
+          id: 'descP',
+          text: getDivContent('description', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('p', forkDiv, {
+          id: 'forkP',
+          text: getDivContent('forks', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('p', updatedDiv, {
+          id: 'updateP',
+          text: getDivContent('updated_at', data, selectedOption),
+          class: 'pClass',
+        });
+        createAndAppend('a', repoDiv, {
+          text: getDivContent('name', data, selectedOption),
+          id: 'repoA',
+          href: getDivContent('html_url', data, selectedOption),
+          target: '_blank',
+        });
+        const selectA = document.getElementById('selectId');
+        selectA.onchange = function func() {
+          selectedOption = document.getElementById('selectId').options[
+            document.getElementById('selectId').selectedIndex
+          ].text;
+          document.getElementById('repoA').href = getDivContent('html_url', data, selectedOption);
+          document.getElementById('repoA').innerText = getDivContent('name', data, selectedOption);
+          changeInnerText('descP', getDivContent('description', data, selectedOption));
+          changeInnerText('forkP', getDivContent('forks', data, selectedOption));
+          changeInnerText('updateP', getDivContent('updated_at', data, selectedOption));
+          if (getDivContent('description', data, selectedOption) === null) {
+            document.getElementById('descDiv').style.display = 'none';
+          } else document.getElementById('descDiv').style.display = 'flex';
+        };
       }
     });
   }
```

Comment:
As a general remark for this function, you are first creating some `p` elements and setting their initial value for the first repo when the application starts. Then, whenever the user changes the selection you update those elements with the new value. While that is a valid approach it can quickly become unwieldy because of the many places that you need to keep track of, by assigning `id` attributes or creating JavaScript variables. Another approach would be just to first clear out any `innerHTML` from the `repoDetailsDiv` (i.e. 'wipe the board clean') and then recreate any necessary inner HTML from scratch. This can result in simpler code.